### PR TITLE
[WIP][BugFix] The max readable version of tablet during schema change should be set as max version

### PR DESCRIPTION
**Pull request by sevev**
If enable sync publish, we will use the max readable version as the tablet version when report the replica version to FE. But if the tablet is doing schema change, the max readable version may never grow before the alter job is finished because these tablets do not have the version before the schema change job and this may cause the alter job to hang. We should skip the version check for these tablets and report the max version as the max readable version.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/tablet_updates.cpp|0|2|00.00%|[2713, 2714]|
|:large_blue_circle:|src/storage/publish_version_manager.cpp|4|4|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 5 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/tablet_updates.cpp|0|2|00.00%|[2713, 2714]|
|:large_blue_circle:|src/storage/publish_version_manager.cpp|0|3|00.00%|[137, 138, 140]|

File: be/src/storage/publish_version_manager.cpp [hunk 1]
```diff
@@ -134,7 +134,11 @@ void PublishVersionManager::update_tablet_version(TFinishTaskRequest& finish_tas
         int64_t tablet_id = tablet_versions[i].tablet_id;
         TabletSharedPtr tablet = StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id);
         if (tablet != nullptr) {
-            tablet_versions[i].__set_version(tablet->max_readable_version());
+            if (tablet->tablet_state() == TabletState::TABLET_RUNNING) {
+                tablet_versions[i].__set_version(tablet->max_readable_version());
+            } else {
+                tablet_versions[i].__set_version(tablet->max_continuous_version());
```
Comment (decster) [inline be/src/storage/publish_version_manager.cpp:140]: max_continuous_version is not max version, doesn't seems to match the description in commit

**Review by decster — COMMENTED** [review]
also, I am not sure after schemachange done, FE can get back to correct version. for example:
during schemachagne, version is 100,
after schemachange, version go back to 90 in BE, but in FE replica version cannot decrease.

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request closed without merging**